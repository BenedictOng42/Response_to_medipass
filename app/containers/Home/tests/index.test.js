import React from 'react';
import { shallow } from 'enzyme';

import { Home, mapDispatchToProps } from '../index';

describe('<Home />', () => {
  const props = {
    updateTable: jest.fn(),
    currentPage: [],
    totalCount: 0,
    loading: false,
    error: false,
  };
  const wrapper = shallow(<Home {...props} />);
  describe('render <Home/>', () => {
    it('should render <Home/> properly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('#componentDidMount', () => {
    it('should run the #updateTable function', () => {
      const instance = wrapper.instance();
      instance.componentDidMount();
      expect(instance.props.updateTable).toBeCalled();
    });
  });
  describe('#mapDispatchToProps', () => {
    it('should dispatch', () => {
      const dispatchSpy = jest.fn();
      const actions = mapDispatchToProps(dispatchSpy);
      Object.keys(actions).forEach((action, index) => {
        actions[action]();
        expect(dispatchSpy).toHaveBeenCalledTimes(index + 1);
      });
    });
  });
});
