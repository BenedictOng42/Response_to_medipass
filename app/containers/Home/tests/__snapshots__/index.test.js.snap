// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render <Home/> should render <Home/> properly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  Pie Of The Day Finder
  <div>
    <WithStyles(SimpleTable)
      error={false}
      headers={
        Array [
          "nameOfPie",
          "price",
          "quantity",
          "storeName",
          "address",
          "rating",
          "contactNumber",
        ]
      }
      items={Array []}
      loading={false}
      onPageChange={[Function]}
      page={1}
      totalCount={0}
    />
  </div>
</div>
`;
